File: c-code/Hv_product.c
#include<stdio.h>
#include<math.h>
// Returns 2D vector
Hv_product_C( int m , int n, float **x , float **h , float **X ) {
	float ax[n][1] ;
	int i , j , k ;
	for( i = 0 ; i < n ; i++ ) {
		ax[i][0] = 0 ;
	}
	for( j = 0 ; j < n ; j++ ) {
		float ax_j = 0 ;
		for( k = 0 ; k < n ; k++ ) {
			float sum1 = 0 ;
			for( i = 0 ; i < m ; i++ ) {
				sum1 + = ( h[i][0] * ( 1 - h[i][0] ) * X[i][j] * X[i][k] ) ;
			}
			ax_j + = ( 1 / m ) * sum1 * x[k][0] ;
		}
		ax[j][0] = ax_j ;
	}
}

File: ex2/Hv_product.m
function ax = Hv_product(x, h, X, m, n)
	ax = zeros(n,1);
	
    for j = 1 : n 
    	ax_j = 0;
		for k = 1: n
			sum1 = 0;
			
			for i = 1 : m
				sum1 = sum1 + (h(i) * (1-h(i)) * X(i,j) * X(i,k));
			end

			% H(j,k) = sum1;
			ax_j = ax_j + (1 / m) * sum1 * x(k);

		end

		ax(j) = ax_j;
	end

end